src/domain/core/src/services/remote_aliases_registry.rs

```diff
@@ -7,29 +7,28 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use async_trait::async_trait;
+use std::sync::Arc;
+
 use internal_error::InternalError;
-use opendatafabric::*;
+use opendatafabric::{AccountName, DatasetName, DatasetPushTarget, RepoName};
 use thiserror::Error;
 
 use crate::*;
 
-#[async_trait]
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
 pub trait RemoteAliasesRegistry: Send + Sync {
     async fn get_remote_aliases(
         &self,
-        dataset_ref: &DatasetRef,
+        dataset: Arc<dyn Dataset>,
```

Comment:
Aliases fundamentally don't belong in an ODF dataset, and likely will be one of the next few things we move into a DB. This will make this interface unusable. Perhaps it should work with `DatasetHandle`s, and for the time being the implementation will use repo to access the file `/<dataset>/info`, but later will switch to DB without breaking the interface.